src/widgets/index/index.ts

```diff
@@ -37,6 +40,40 @@ export type Index = Widget & {
   dispose(options: DisposeOptions): void;
 };
 
+function isIndexWidget(widget: Widget): widget is Index {
+  return widget.$$type === 'ais.index';
+}
+
+function resolveScopedResultsFromWidgets(widgets: Widget[]): ScopedResult[] {
+  const indexWidgets = widgets.filter(isIndexWidget);
+
+  if (indexWidgets.length === 0) {
```

Comment:
This branch is useless since we reduce below. The returned value will be the initial state.